File: myurlshortner-consumer/myurlshortner-consumer/src/main/scala/com/acme/myurlshortner/consumer/domain/entity/useragent/Device.scala
package com.acme.myurlshortner.consumer.domain.entity.useragent

sealed trait Device(
    val value: String,
    val userAgentValue: String | Nothing
)

case object PC      extends Device("PC", null)
case object Mac     extends Device("Macintosh", "CPU OS")
case object iPad    extends Device("iPad", "CPU iPad")
case object iPhone  extends Device("iPhone", "CPU iPhone")
case object Android extends Device("Android", "Android")

File: myurlshortner-consumer/src/main/kotlin/com/acme/myurlshortner/consumer/domain/useragent/Device.kt
package com.acme.myurlshortner.consumer.domain.useragent

sealed class Device {
    val userAgentValue: String?

    constructor(userAgentValue: String?) {
        this.userAgentValue = userAgentValue
    }

    data object PC : Device("PC")
    data object Mac : Device("Mac")
    data object iPad : Device("iPad")
    data object iPhone : Device("iPhone")
    data object Android : Device("Android")
    data object Other : Device(null)
}
